### ISSUE TITLE
Map implementation doesn't support constant time lookup

### ISSUE THREAD

**Issue by rtheunissen** [php-ds/polyfill#13]
If anyone would like to implement an object-key supported insertion-ordered constant lookup hash table in PHP let me know. :snail:

**Comment by rvanvelzen** [comment]
Having given this some thought, I believed this to be *relatively* straightforward to implement.

In [6a380807](https://github.com/rvanvelzen/php-collection/commit/6a38080774024cc3b37fcc75553e386931efe2ac) I attempted to do su using hash-buckets and a doubly-linked list.

However… After some extensive testing with >30.000 elements added, it only produced segfaults. Turns out that much references to objects causes the garbage collector to go crazy.

I don't believe there is a more usable memory-efficient way to implement this, so I gave up on the idea and [dropped keeping insertion-order](https://github.com/rvanvelzen/php-collection/commit/d4ef53f2672358d5ef279eb4cbead0b3d7b88220).

Hope this is some kind of useful :)
